File: ios/Plugin/OverlaySettings.swift
import Foundation;

enum OverlaySettingsError: Error {
    case invalidArgument(message: String)
}

public class OverlaySettings: NSObject
{
    private var path: URL;
    private var top: Int = 0;
    private var left: Int = 0;
    private var height: Int = 0;
    private var width: Int = 0;
    private var fillBehaviour: String = "none";
    private var opacity: Float = 1;
    
    init(path: URL, top: Int, left: Int, height: Int, width: Int, fillBehaviour: String, opacity: Float) throws
    {
        
        if (height < 0)
        {
            throw OverlaySettingsError.invalidArgument(message: "Parameter height cannot be negative");
        }
        if (width < 0)
        {
            throw OverlaySettingsError.invalidArgument(message: "Parameter width cannot be negative");
        }
        
        self.path = path;
        self.height = height;
        self.width = width;
        self.top = top;
        self.left = left;
        self.fillBehaviour = fillBehaviour;
        self.opacity = opacity;
    }
    
    func getPath()->URL
    {
        return self.path;
    }
    
    func getHeight()->Int
    {
        return self.height;
    }
    
    func setHeight(_ height: Int) throws
    {
        if (height < 0)
        {
            throw OverlaySettingsError.invalidArgument(message: "Parameter height cannot be negative");
        }
        self.height = height;
    }
    
    func getWidth()->Int
    {
        return self.width;
    }
    
    func setWidth(_ width: Int) throws
    {
        if (width < 0)
        {
            throw OverlaySettingsError.invalidArgument(message: "Parameter width cannot be negative");
        }
        self.width = width;
    }
    
    
    func getTop()->Int
    {
        return self.top;
    }
    
    func setTop(_ top: Int) throws
    {
        self.top = top;
    }

    func getLeft()->Int
    {
        return self.left;
    }
    
    func setLeft(_ left: Int) throws
    {
        self.left = left;
    }

    func getFillBehaviour()->String
    {
        return self.fillBehaviour;
    }
    
    func setFillBehaviour(_ fill: String) throws
    {
        self.fillBehaviour = fill;
    }
    
    
    func getOpacity()->Float
    {
        return self.opacity;
    }
    
    func setOpacity(_ opacity: Float) throws
    {
        self.opacity = opacity;
    }
    
}

File: android/src/main/java/com/whiteguru/capacitor/plugin/videoeditor/OverlaySettings.java
package com.whiteguru.capacitor.plugin.videoeditor;

public class OverlaySettings {

    private String path = "";
    private int top = 0;
    private int left = 0;
    private int height = 0;
    private int width = 0;
    private String fillBehaviour = "none";
    private float opacity = 1.0f;

    public OverlaySettings() {
    }

    public OverlaySettings(String path, int top, int left, int height, int width, String fillBehaviour, float opacity) {
        if (height < 0) {
            throw new IllegalArgumentException("Parameter height cannot be negative");
        }

        if (width < 0) {
            throw new IllegalArgumentException("Parameter width cannot be negative");
        }
        this.path = path;
        this.top = top;
        this.left = left;
        this.height = height;
        this.width = width;
        this.fillBehaviour = fillBehaviour;
        this.opacity = opacity;
    }

    public final String getPath(){ return path; }
    public void setPath(String path){
      this.path = path;
    }

    public int getTop() {
      return top;
    }
    public void setTop(int top) {
      this.top = top;
    }

    public int getLeft() {
      return left;
    }
    public void setLeft(int left) {
      this.left = left;
    }

    public int getHeight() {
        return height;
    }

    public void setHeight(int height) {
        if (height < 0) {
            throw new IllegalArgumentException("Parameter height cannot be negative");
        }

        this.height = height;
    }

    public int getWidth() {
        return width;
    }

    public void setWidth(int width) {
        if (width < 0) {
            throw new IllegalArgumentException("Parameter width cannot be negative");
        }

        this.width = width;
    }

    public final String getFillBehaviour(){ return fillBehaviour; }
    public void setFillBehaviour(String fillBehaviour){
      this.fillBehaviour = fillBehaviour;
    }

    public float getOpacity() {
      return opacity;
    }
    public void setOpacity(float opacity) {
      this.opacity = opacity;
    }


}
